### DISCUSSION TITLE
I'd like to download and run... but couldn't figure out how (solved)

### DISCUSSION THREAD

**Discussion by CleverCoder** [ApicurioArchive/apicurio-studio#2281] [General]
_(Edited to prevent ongoing negative conversation. The original comments were unprofessional and embarrassing)_

Hey! So, I'm used to using Stoplight studio, but they changed things so you must login and create an account. I understand the need to monetize, and the value of collaboration features, but this change led me to look for alternatives.

I tried to download and run Apicurio but couldn't get it working.

Would it be possible to update the documentation to help guide future users through setting up and running Apicurio? I'd like to see what it can do, but don't have the time to sort through the issues I've been facing.

Thank you for all the hard work. I'll be giving it another go in the future.

Cheers!
- Sean

_(PS... I did manage to run it using the "standalone.sh", but it wasn't clear what I needed to do from that point)_

**Comment by EricWittmann** [comment]
Normally I might ignore something like this, given the needlessly aggressive tone.  I mean, is the 100% free software we've written not easy enough for you to consume?

But we have been neglecting Studio a lot lately (as evidenced by how long it has taken me to respond to this), and there is some understandable community frustration I think.  Maybe this particular case isn't all that understandable, but still.

So for anyone in the future who sees this and just wants a simple editing tool without having to log into a cloud instance or install something locally, we've had a thing called Apicurito for quite some time:

https://www.apicur.io/apicurito/

Too bad you weren't able to find that.  In any case, you can try it live here:

https://www.apicur.io/apicurito/pwa/

Also note:  it's a Portable Web Application, which means that if you're using Chrome you can "install" it locally:

![image](https://github.com/Apicurio/apicurio-studio/assets/1890703/d90b8fff-f807-4e6b-ae6f-ee70874a0439)

See the little install icon next to the star icon in the address bar.

**Reply by EricWittmann** [reply]
Also, for posterity, you can find installation documentation here:

https://www.apicur.io/studio/docs/

Maybe it's not enough, but lots of people have been successful with it, so I guess ymmv.

**Reply by Rudxain** [reply]
> aggressive tone

How? I agree it was unprofessional, but "aggressive"? I'm genuinely curious about why some people felt it as that

**Reply by msavy** [reply]
> > aggressive tone
> 
> How? I agree it was unprofessional, but "aggressive"? I'm genuinely curious about why some people felt it as that

The original comment has been edited. It has been resolved now, no further comment needed.

**Reply by Rudxain** [reply]
> The original comment has been edited

I've read both the original (in the commit history) and the edited versions. The original does have a little bit of sarcasm, which is "passive-aggressive" instead of "aggressive". Everything else seems like regular "complaining tone" for me

**Reply by msavy** [reply]
Please, the situation has been resolved. No additional messages are required.

**Comment by msavy** [comment]
> I did manage to run it using the "standalone.sh" - which is sad, as API designers shouldn't have to be bothered with what a "shell" script is -

I can't believe an IT professional wrote this 🤣🤣.

**Reply by CleverCoder** [reply]
Granted, the wording was short and didn't convey the proper sentiment, and the tone was not great. For that, I apologize.

I use tons of shell scripts each and every day, but for a tool that appeared to be something I can run locally, I find  that the better user experience is to have a native platform launch app bundle, exe, etc... My complaint wasn't the use of the shell script, but the overall process of getting it running.

A professional knows that tools that "just work" out of the box are the ones that get used. Debugging scripts and build processes is a barrier to entry that will make those of us who just have a job to get done move on to something else. When your goal is to maintain API contracts, and you can't download and run software that's supposed to help with that, then what? This was approaching the software from a purely objective "willing to give it a chance". I couldn't get it running within 30 minutes, so I wanted to share my experience.
Professionals also know that time is money.

If I was on the team that maintains this, I would be interested in new users' experiences. I'm sorry if it came across in any negative way, but seriously. I tried it... looked for documentation on how to get it running... couldn't find it, left a note, and moved on with my day.
I'll likely circle back and try the links above or give it another shot in the future.

@EricWittmann - thank you for the response, and additional information. That is very helpful.

**Reply by RobThree** [reply]
> Professionals also know that time is money.

"Professionals" also don't write in a sarcastic tone (_"seems interesting, as it's "open source""_), don't confuse objective and subjective opinions (_"misleads into thinking it's easy to get up and running"_), don't call a gifted horse "rubbish", aren't asking in a demanding tone of voice _"Can you fix this"_, don't complain like entitled litte brats (_"I don't want to be bothered with..."_), don't demand (_"Give me a tool that"_), don't dictate (_"If it's a web-based tool, then [...blahblahblah]... like VSCode."_), don't say stuff like _"I mean... cmon"_.

I'm dutch. I've been called an a-hole on many occasion because we don't beat around the bush, don't wrap everything in niceties and get straight to the point. It's a culture thing. So I get it when someone thinks I'm a bit of a jerk. But, holy moly, next time I'll point them to this issue and show them what a real self-centred, entitled, jerk looks like.

Also: What's stopping you from forking and improving? Maybe even make it into an PR?

_I mean... cmon._

**Reply by CleverCoder** [reply]
I updated the message after re-reading, as I agree. It was clearly written likely late at night, and under less than ideal emotional state. Not the finest moment. Who was that guy anyway (it must have been a bad day) 🤦‍♂️ 
Everyone's remarks are justified. I appreciate the hard work that goes into projects like this.

**Reply by EricWittmann** [reply]
Everyone has a bad day, and rest assured I didn't lose any sleep over the post.  :)  I'll close the discussion.
